portability-api/src/main/java/org/dataportabilityproject/webapp/StartCopyHandler.java

```diff
@@ -123,31 +121,47 @@ private void handleWorkerAssignmentFlow(HttpExchange exchange, String id)
     // We have the data, now update it unassigned so it can be assigned a worker
     // Set Job to state to pending worker assignment
     jobDao.updateJobStateToPendingWorkerAssignment(job.id()); // Now that job is complete unassiged
+    logger.debug("Updated updateJobStateToPendingWorkerAssignment, id: {}", job.id());
 
     // Loop until the worker updates it to assigned without auth data state, e.g. at that point
     // the worker instance key will be populated
     // TODO: start new thread
     // TODO: implement timeout condition
     // TODO: Handle case where API dies while waiting
     while (jobDao.lookupAssignedWithoutAuthDataJob(job.id()) == null) {
+      logger.debug("No result for lookupAssignedWithoutAuthDataJob, id: {}", job.id());
       try {
-        Sleeper.DEFAULT.sleep(5000);
+        Sleeper.DEFAULT.sleep(10000);
       } catch (InterruptedException e) {
         throw new RuntimeException(e);
       }
     }
 
+    logger.debug("Found job after while loop, lookupAssignedWithoutAuthDataJob, id: {}", job.id());
+
     // Ensure job is assigned and has worker key
     PortabilityJob assignedJob = jobDao.lookupAssignedWithoutAuthDataJob(job.id());
+
+    logger.debug("Found job after lookupAssignedWithoutAuthDataJob, id: {}", job.id());
     Preconditions.checkNotNull(assignedJob.workerInstancePublicKey() != null);
     // Populate job with auth data from cookies encrypted with worker key
-    PublicKey publicKey = PublicPrivateKeyUtils
+    logger.debug("About to parse: {}", assignedJob.workerInstancePublicKey());
+    PublicKey publicKey = PublicPrivateKeyPairGenerator
         .parsePublicKey(assignedJob.workerInstancePublicKey());
-    jobDao.updateJobStateToAssigneWithAuthData(assignedJob.id(),
-        cryptoHelper.encryptAuthData(publicKey, exportAuthCookie),
-        cryptoHelper.encryptAuthData(publicKey, importAuthCookie));
+    logger.debug("Found publicKey: {}", publicKey.getEncoded().length);
+
+    String encryptedExportAuthData = cryptoHelper.encryptAuthData(publicKey, exportAuthCookie);
```

Comment:
at this point is it the whole cookie or just credential that we're encrypting?

Reply:
Good point, I clarified the variable name. It's just the cookie value, not the whole cookie.